addons/website_forum/tests/test_forum_karma_access.py

```diff
@@ -183,6 +183,20 @@ def test_ask_question(self):
         })
         self.assertEqual(self.user_portal.karma, KARMA['post'] + KARMA['gen_que_new'], 'website_forum: wrong karma generation when asking question')
 
+        # Try to bypass karma check with context values.
+        self.forum.karma_dofollow = self.user_portal.karma + 10
+        post = Post.with_user(self.user_portal).with_context(default_content='<p>Super <a href="www.link.com">Link</a></p>').create({
+            'name': "Bypass",
+            'forum_id': self.forum.id,
+        })
+        self.assertTrue("nofollow" in post.content, 'website_forum: default_content in context should not bypass karma check.')
+
+        post = Post.with_user(self.user_portal).with_context(default_forum_id=self.forum.id).create({
+            'name': "Bypass",
+            'content': '<p>Super <a href="www.link.com">Link</a></p>',
+        })
+        self.assertTrue("nofollow" in post.content, 'website_forum: default_forum_id in context should not bypass karma check.')
```

Comment:
Should also test that nofollow is NOT in content if user has enough karma, to make sure we didn't break anything.
(Unless already done somewhere).

Reply:
Note that technically you have more use cases, because you can combine both context and both direct values too, not sure if worth testing everything though, could maybe be included in a loop with some nice subTest cases?